couch/test_couch.py

```diff
@@ -121,3 +102,117 @@ def test_couch_blacklist(self):
                     self.assertMetric(gauge, tags=tags, count=0)
                 else:
                     self.assertMetric(gauge, tags=tags, count=1)
+
+@attr(requires='couch')
+@attr(couch_version='2.x')
+class TestCouchdb2(AgentCheckTest):
+    """Basic Test for couchdb2 integration."""
+    CHECK_NAME = 'couch'
+
+    NODE1 = {
+        'server': 'http://127.0.0.1:5984',
+        'user': 'dduser',
+        'password': 'pawprint',
+        'name': 'node1@127.0.0.1'
+    }
+
+    NODE2 = {
+        'server': 'http://127.0.0.1:5984',
+        'user': 'dduser',
+        'password': 'pawprint',
+        'name': 'node2@127.0.0.1'
+    }
+
+    NODE3 = {
+        'server': 'http://127.0.0.1:5984',
+        'user': 'dduser',
+        'password': 'pawprint',
+        'name': 'node3@127.0.0.1'
+    }
+
+    def __init__(self, *args, **kwargs):
+        AgentCheckTest.__init__(self, *args, **kwargs)
+        self.cluster_gauges = []
+        self.by_db_gauges = []
+        with open('couch/metadata.csv', 'rb') as csvfile:
+            reader = csv.reader(csvfile)
+            reader.next() # This one skips the headers
+            reader.next() # This and the next one skip CouchDB 1.x specific metrics not included in 2.x
```

Comment:
I would not rely on the order of the metadata file to ensure that the test works, unfortunately. Can you specifically filter these out?